### PR TITLE
Tickets/DM-29264: update filter header

### PR DESCRIPTION
**Pull request by suberlak**
Update comcam filter header to conform to information in https://github.com/lsst/obs_lsst/blob/master/python/lsst/obs/lsst/filters.py  - LsstCam accepts phosim filter names for gen3 butler repository (ugrizy), but for LsstComCam, these are updated to `g_01`, etc.

### CODE REVIEW

File: python/lsst/phosim/utils/phosim_repackager.py [hunk 1]
```diff
@@ -40,6 +40,41 @@ def noao_section_keyword(bbox, flipx=False, flipy=False):
     return "[%i:%i,%i:%i]" % (xmin, xmax, ymin, ymax)
 
 
+def translate_comcam_filter(phosimFilter):
+    """
+    Translate phosim filter name to that appropriate
+    for obs_lsst - see DM-21706
```
Comment (teweitsai) [inline python/lsst/phosim/utils/phosim_repackager.py]: The period in the end of sentence.

File: python/lsst/phosim/utils/phosim_repackager.py [hunk 2]
```diff
@@ -40,6 +40,41 @@ def noao_section_keyword(bbox, flipx=False, flipy=False):
...
+
+    Parameters
+    ----------
+    phosimFilter: str
+        Name of phosim filter in the phosim file header,
+        one of u,g,r,i,z,y
```
Comment (teweitsai) [inline python/lsst/phosim/utils/phosim_repackager.py]: The period in the end of sentence.

File: python/lsst/phosim/utils/phosim_repackager.py [hunk 3]
```diff
@@ -40,6 +40,41 @@ def noao_section_keyword(bbox, flipx=False, flipy=False):
...
+
+    Returns
+    --------
+    mappedFilter: str
+        A translated filter name allowed by obs_lsst
+        for lsstComCam instrument
```
Comment (teweitsai) [inline python/lsst/phosim/utils/phosim_repackager.py]: The period in the end of sentence.

File: python/lsst/phosim/utils/phosim_repackager.py [hunk 4]
```diff
@@ -40,6 +40,41 @@ def noao_section_keyword(bbox, flipx=False, flipy=False):
...
+        "y": "04",
+    }
+    # test if filter can be translated
+    if phosimFilter in phosim_to_comcam.keys():
+        suffix = phosim_to_comcam[phosimFilter]
+        return f"{phosimFilter}_{suffix}"
+    else:
+        raise ValueError(
+            f"This phosim filter:{phosimFilter} cannot \
+            be translated for comcam")
```
Comment (teweitsai) [inline python/lsst/phosim/utils/phosim_repackager.py:83]: Please document the `ValueError` in function's doc string as well.

File: python/lsst/phosim/utils/phosim_repackager.py [hunk 5]
```diff
@@ -158,6 +193,14 @@ def repackage(self, phosim_amp_files, out_dir="."):
...
             )
 
+            # Set the filter information.
+            # For lsstCam, the phosim filter names (ugrizy)
+            # are appropriate, but for comCam they need
+            # to have a different name as in obs_lsst filters.py
+            if self.instrument == "comCam":
+                phosimFilter = hdu.header["FILTER"]
+                hdu.header["FILTER"] = translate_comcam_filter(phosimFilter)
+
```
Comment (teweitsai) [inline python/lsst/phosim/utils/phosim_repackager.py]: I felt this part of code is duplicated in `repackage()` and `repackage_eimage()`. Do you want to consider to abstract this such as `updateComCamSpecificData(header)` and put the related specific setting/data of ComCam into this function to reduce the duplication of code (this also means you could remove the `translate_comcam_filter()` and put the logic into the `updateComCamSpecificData(header)`. The benefit is that you will only need to modify one function if you find there would be other data needs to change in additional to filter in ComCam.

**Review by teweitsai — APPROVED** [review]
This PR looks good to me. Please consider my comments before the merge of code. Thanks!

File: python/lsst/phosim/utils/phosim_repackager.py (hunk 4 above)
Comment (suberlak) [reply python/lsst/phosim/utils/phosim_repackager.py:83]: Added documentation

File: python/lsst/phosim/utils/phosim_repackager.py (hunk 5 above)
Comment (suberlak) [reply python/lsst/phosim/utils/phosim_repackager.py]: Moved from `translate_comcam_filter(phosimFilter)` to  `updateComCamSpecificData(header)` function  that takes the FITS header and does appropriate updates (currently updating the FILTER header). Added appropriate test.

**Pull request merged**